### PR TITLE
[QI2-1223] Call reusable authentication script for e2e

### PR DESCRIPTION
**Pull request by oschusler**

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -79,10 +83,28 @@ jobs:
...
+          git switch feature/QI2-1222/include-auth-script
+          pwd
+      - name: Get authentication token
+        run: |
+          python -m venv env
+          . ./env/bin/activate
+          mkdir -p ~/.quantuminspire
+          pip install -r requirements.txt
+          python get_oauth_tokens.py > ~/.quantuminspire/config.json
+          cat ~/.quantuminspire/config.json
```
Comment (eliasKA) [inline .github/workflows/tests.yaml:99]: Why are we doing the `cat` here? Do we want to expose this data in the action logs?

File: .github/workflows/tests.yaml [hunk 2]
```diff
@@ -79,10 +83,28 @@ jobs:
...
+          deactivate
+        working-directory: GitHub-Actions/scripts/oauth_tokens
       - name: Install packages
         run: poetry install
       - name: Run e2e test
-        run: poetry run python tests/e2e_test.py "${{ matrix.backend-name }}"
+        run: |
+          cat ~/.quantuminspire/config.json
```
Comment (eliasKA) [inline .github/workflows/tests.yaml:106]: No need to do it here again if we're doing it in the previous step already WDYT?

**Review by eliasKA — CHANGES_REQUESTED** [review]

File: tests/e2e_test.py [hunk 3]
```diff
@@ -98,7 +98,7 @@ def _run_e2e_tests(name: str) -> None:
 
 
 def main(name: str) -> None:
-    _get_auth_tokens()
+    # _get_auth_tokens()
```
Comment (NischalQuTech) [inline tests/e2e_test.py:101]: commented out code ?

File: .github/workflows/tests.yaml [hunk 4]
```diff
@@ -69,6 +69,10 @@ jobs:
     env:
       E2E_USERNAME:  ${{ secrets.E2E_USERNAME }}
       E2E_PASSWORD:  ${{ secrets.E2E_PASSWORD }}
+      IDP_URL: https://quantum-inspire-staging.eu.auth0.com
+      IDP_CLIENT_ID_E2E_TEST: Yz7ni9PUAyT43eUASZfmc1yqI66QxLUJ
+      COMPUTE_JOB_MANAGER_API_AUDIENCE: compute-job-manager
+      QI2_default_host: https://staging.qi2.quantum-inspire.com
```
Comment (NischalQuTech) [inline .github/workflows/tests.yaml:75]: are we planning to keep this hardcoded here or is this supposed to be derived from secrets ?

**Pull request closed without merging**